### ISSUE TITLE
Validate that data output by sources matches declared data type

### ISSUE THREAD

**Issue by cgardens** [airbytehq/airbyte#1421]
## Tell us about the problem you're trying to solve
* One of the reasons we see normalization fail is that the value provided by the source is not actually of the type that the source declare it was. It should should not be the responsibility of the destination to verify that sources are outputting the data they said they would.

## Describe the solution you’d like
* Sync worker should verify that the data output by the source matches the type for the declared field. (e.g. a number is in fact a number and not a string).

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368137828128) by [Unito](https://www.unito.io)

**Comment by ChristopheDuong** [comment]
An example of this is trying to run normalization on top of a Freshdesk sync with out integration tests account for:
https://github.com/airbytehq/airbyte/issues/3180

in the catalog.json of the `companies` stream, I see:
  ```
  "health_score": {
      "type": [
          "number",
          "null"
      ]
  },
  ```

But I see in the logs:
  
  ```
  2021-05-03 13:54:37 INFO (/tmp/workspace/3/0) LineGobbler(voidCall):69 - [33mDatabase Error in model companies_b14 (models/generated/airbyte_tables/data/companies_b14.sql)[0m
  2021-05-03 13:54:37 INFO (/tmp/workspace/3/0) LineGobbler(voidCall):69 -   invalid input syntax for type double precision: "Doing okay"
  2021-05-03 13:54:37 INFO (/tmp/workspace/3/0) LineGobbler(voidCall):69 -   compiled SQL at ../build/run/airbyte_utils/models/generated/airbyte_tables/data/companies_b14.sql
  ```

Double-checking in the raw table of my destination:
![Screenshot 2021-05-03 at 16 05 55](https://user-images.githubusercontent.com/7718175/116886611-934f0c80-ac29-11eb-8716-9a527d3e0c7c.png)

**Comment by sherifnada** [comment]
this was solved in #3253
